src/foam/u2/Element.js

```diff
@@ -1248,14 +1245,17 @@ foam.CLASS({
 
     function removeAttribute(name) {
       /* Remove attribute named 'name'. */
-      for ( var i = 0 ; i < this.attributes.length ; i++ ) {
-        if ( this.attributes[i].name === name ) {
-          this.attributes.splice(i, 1);
-          delete this.attributeMap[name];
-          this.onRemoveAttr(name);
-          return;
-        }
-      }
+
+      // NOTE: This call might throw an exception if it happens to be called on
+      // an element that's in the OUTPUT state. If that's the case, we don't
+      // want to remove the entry in `this.attributes` or `this.attributeMap`,
+      // otherwise when the element is in the LOADED state later and we try to
+      // remove the attribute, the attribute will have been removed from the
+      // array and map, but the attribute will still be set on the DOM element.
+      this.onRemoveAttr(name);
```

Comment:
We have this same sort of ordering issue for all methods like "removeChild" which calls onRemoveChild.  I'm not sure we should bother to re-order all of them.  Calling anything but load() in the OUTPUT state is an error.

Additionally I'm not wild about the idea of calling onRemoveAttr() before we've actually updated our state, it feels wrong.  I see why you did it, so that the exception triggers before we've updated state and we can therefore successfully remove the attribute on a later attempt, but I don't think it's worth it.  This feels like a bandaid over a deeper bug of whatever code is outputting elements but not calling .load() immediately afterwards, such code shouldn't exist, and if it does we should figure out why.

Reply:
Ok, I'll try to find out what's calling `.load` at the wrong time.

Reply:
@adamvy Fixed. I reverted the commit that made the change you commented on and found a different way to avoid the situation happening in the first place. See 5b1c62d for the fix.